### PR TITLE
[codex] Refactor LLM reply continuation into agent runs

### PR DESCRIPTION
**Pull request by eanzhao**
## Summary
- Implements issue #596 Phase A by replacing the channel LLM reply hosted inbox singleton with run-scoped `AgentRunGAgent` continuation actors.
- Adds `IChannelLlmReplyRunDispatcher` and the NyxidChat dispatcher that creates/gets `AgentRunGAgent` by `correlationId` and dispatches typed `AgentRunStartRequested` commands.
- Moves the existing LLM reply gates, metadata enrichment, fallback timeout, streaming sink, reply generation, ready/drop dispatch, and relay token echo into the run actor while keeping reply tokens out of persisted run state/events.
- Updates `ConversationGAgent` and tests to use the run dispatcher path and retires the old inbox interface/hosted service path.

## Validation
- `dotnet build test/Aevatar.GAgents.ChannelRuntime.Tests/Aevatar.GAgents.ChannelRuntime.Tests.csproj --nologo`
- `dotnet test test/Aevatar.GAgents.ChannelRuntime.Tests/Aevatar.GAgents.ChannelRuntime.Tests.csproj --nologo`
- `dotnet test test/Aevatar.GAgents.Channel.Protocol.Tests/Aevatar.GAgents.Channel.Protocol.Tests.csproj --nologo`
- `bash tools/ci/test_stability_guards.sh`
- `bash tools/ci/architecture_guards.sh`

Refs #596

### CODE REVIEW

File: agents/Aevatar.GAgents.NyxidChat/AgentRunDispatcher.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+            Timestamp = Timestamp.FromDateTimeOffset(_timeProvider.GetUtcNow()),
+            Payload = Any.Pack(command),
+            Route = EnvelopeRouteSemantics.CreateDirect("channel-llm-reply-run-dispatcher", actor.Id),
+            Propagation = new EnvelopePropagation
+            {
+                CorrelationId = runId,
+            },
+        };
+
+        await _actorDispatchPort.DispatchAsync(actor.Id, envelope, ct);
```
Comment (eanzhao) [inline agents/Aevatar.GAgents.NyxidChat/AgentRunDispatcher.cs]: **[codex]** severity=blocker, category=concurrency

This uses synchronous `IActorDispatchPort.DispatchAsync` to start the run actor. `ConversationGAgent` awaits this while handling the inbound turn, and `AgentRunGAgent` later dispatches `LlmReplyReadyEvent` back to that same conversation actor, so the real Local/Orleans dispatch implementations can form an A -> B -> A wait cycle and also block the webhook for the full LLM run. This violates CLAUDE.md: cross-actor waits must be continuationized. Start the run through a one-way enqueue/publish mechanism that returns after acceptance, then let the run actor deliver the ready/drop signal independently.

File: agents/Aevatar.GAgents.NyxidChat/AgentRunGAgent.cs [hunk 2]
```diff
@@ -57,109 +66,106 @@ public ChannelLlmReplyInboxRuntime(
...
+        return ActorIdPrefix + correlationId.Trim();
     }
 
-    public async Task StopAsync(CancellationToken ct)
+    protected override AgentRunGAgentState TransitionState(AgentRunGAgentState current, IMessage evt) =>
+        StateTransitionMatcher
+            .Match(current, evt)
+            .On<AgentRunStartedEvent>(ApplyStarted)
+            .On<AgentRunReplyProducedEvent>(ApplyReplyProduced)
+            .On<AgentRunDroppedEvent>(ApplyDropped)
```
Comment (eanzhao) [inline agents/Aevatar.GAgents.NyxidChat/AgentRunGAgent.cs:86]: **[glm-5.1]** severity=major, category=arch

`AgentRunGAgent` reaches terminal states (`ReplyProduced`, `Dropped`, `Failed`) but has no self-deactivation or cleanup mechanism. Per AGENTS.md: *"默认优先 run/session/task-scoped actor"* and *"长期 actor 只保留给事实拥有者"*. Each LLM reply creates a persistent actor that accumulates indefinitely. Consider persisting the terminal event then scheduling a self-deactivation (e.g. `IActorRuntime.DeactivateAsync` or a delayed cleanup self-message) so the actor does not remain alive after its single-shot purpose is served.

File: agents/Aevatar.GAgents.NyxidChat/AgentRunGAgent.cs [hunk 3]
```diff
@@ -57,109 +66,106 @@ public ChannelLlmReplyInboxRuntime(
...
+        var runId = NormalizeOptional(request.CorrelationId) ?? Id;
+        var startedAtUnixMs = _timeProvider.GetUtcNow().ToUnixTimeMilliseconds();
 
-    public async ValueTask DisposeAsync()
-    {
-        await StopAsync(CancellationToken.None);
-    }
-
-    internal const long MaxInboxRequestAgeMs = 5 * 60 * 1000;
+        if (State.Status is AgentRunStatus.ReplyProduced or AgentRunStatus.Dropped or AgentRunStatus.Failed)
```
Comment (eanzhao) [inline agents/Aevatar.GAgents.NyxidChat/AgentRunGAgent.cs:104]: **[Consensus: 2 models]** severity=major, category=bug

The terminal-state dedup guard at this line silently drops undelivered ready/drop notifications when conversation dispatch fails. The actor persists `AgentRunReplyProducedEvent` / `AgentRunFailedEvent` / `AgentRunDroppedEvent` *before* dispatching to the conversation actor; if that dispatch throws, a retry of the start command will hit this guard and return without re-emitting the signal — `ConversationGAgent.PendingLlmReplyRequests` stays stuck. The guard also has zero test coverage. Either move the guard to *after* a confirmed-delivered phase (dispatch retry as a persisted phase), or only mark the run terminal once the conversation acknowledges. Add a test covering both the legitimate-duplicate case and the failed-dispatch retry case.

<details>
<summary>Per-model verbatim</summary>

- **kimi**: `HandleStartAsync` introduces a new terminal-state dedup guard (ignoring duplicate start commands when the run is already `ReplyProduced`, `Dropped`, or `Failed`). This idempotency behavior is not covered by any test in `AgentRunGAgentTests.cs`. Add a test that sends a second `AgentRunStartRequested` after the run reaches a terminal state and asserts it is ignored.
- **codex**: This terminal duplicate guard can drop undelivered results. The run persists `AgentRunReplyProducedEvent` / `AgentRunFailedEvent` before `DispatchReadyEventAsync`, and persists `AgentRunDroppedEvent` before the drop notification; if that conversation dispatch fails, a retry of the start command sees terminal state here and returns without re-emitting the ready/drop signal, leaving `ConversationGAgent.PendingLlmReplyRequests` stuck. Make delivery a retryable persisted phase, or only mark the run terminal after the conversation has accepted/acknowledged the notification.
</details>

File: agents/Aevatar.GAgents.NyxidChat/AgentRunGAgent.cs [hunk 4]
```diff
@@ -57,109 +66,106 @@ public ChannelLlmReplyInboxRuntime(
...
-        if (_subscription is not null)
-            return;
-
-        _subscription = await _streamProvider
-            .GetStream(InboxStreamId)
-            .SubscribeAsync<NeedsLlmReplyEvent>(ProcessAsync, ct);
-
-        _logger.LogInformation("Started channel LLM reply inbox on {StreamId}", InboxStreamId);
+        ArgumentException.ThrowIfNullOrWhiteSpace(correlationId);
+        return ActorIdPrefix + correlationId.Trim();
```
Comment (eanzhao) [inline agents/Aevatar.GAgents.NyxidChat/AgentRunGAgent.cs:78]: **[glm-5.1]** severity=minor, category=arch

`HandleStartAsync` calls `PersistDomainEventAsync(AgentRunStartedEvent)` then synchronously calls `ProcessAsync` which runs the full LLM generation (up to 300s). If `PersistDomainEventAsync` for the started event succeeds but `ProcessAsync` throws an unguarded exception (e.g. from `EnsureTargetActorAsync` or a subsequent `PersistDomainEventAsync` call), the actor is left in `Started` state with no terminal event and no retry path. Wrap the `ProcessAsync` call in a try/catch that persists `AgentRunFailedEvent` on any unexpected exception to guarantee terminal state is always reached.

File: test/Aevatar.GAgents.ChannelRuntime.Tests/AgentRunGAgentTests.cs [hunk 5]
```diff
@@ -712,6 +755,8 @@ private sealed class DispatchingActorRuntime(params (string Id, IActor Actor)[]
             static pair => pair.Actor,
```
Comment (eanzhao) [inline test/Aevatar.GAgents.ChannelRuntime.Tests/AgentRunGAgentTests.cs:1028]: **[glm-5.1]** severity=minor, category=test

`NoOpEventSourcing.TransitionState` returns `current` unchanged, so `AgentRunGAgent.State` never advances past `AGENT_RUN_STATUS_UNSPECIFIED` in any test. This means the duplicate-terminal-start guard at `HandleStartAsync` line ~78 (`State.Status is AgentRunStatus.ReplyProduced or AgentRunStatus.Dropped or AgentRunStatus.Failed`) is never exercised. Either wire a real `IEventSourcingBehavior` that calls `TransitionState`, or add a dedicated test that pre-populates the state to a terminal status and asserts the guard fires.
